content/en/blog/_posts/2025-XX-XX-endpoints-deprecation.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: blog
+title: "Endpoints formally deprecated in favor of EndpointSlices"
+slug: endpoints-deprecation
+date: 2025-XX-XX
+draft: true
+author: >
+  Dan Winship (Red Hat)
+---
+
+Since the addition of [EndpointSlices] several years ago, the
+Endpoints API in Kubernetes has been gathering dust. New Service
+features like [dual-stack networking] and [topology-aware routing] are
+only supported via the EndpointSlice API, so all service proxies,
+Gateway API implementations, and similar controllers have had to be
+ported from using Endpoints to using EndpointSlices. At this point,
+the Endpoints API is really only there to avoid breaking end user
+workloads and scripts that still make use of it.
+
+We are now beginning the process of officially deprecating the
+Endpoints API, with clarifications to the documentation, and warnings
+from the API server when a client reads or writes Endpoints resources.
+
+While the Endpoints type itself can never go away as long as the "v1"
```

Comment:
```suggestion
While the Endpoints type itself can never go away as long as the `v1` Endpoints
```

We technically can drop Endpoints v1 without dropping all v1 APIs (would need a major Kubernetes version, but if we ever do a major version bump, our compatibility guarantees allow doing anything we like with the APIs).

Reply:
That basically says "Endpoints can't go away as long as Endpoints hasn't gone away".

I couldn't come up with an entirely correct explanation that didn't get mired in irrelevant details, so I just changed it to say Endpoints will "probably" never go away.